content/helpers.go

```diff
@@ -206,3 +206,29 @@ func copyWithBuffer(dst io.Writer, src io.Reader) (written int64, err error) {
 	bufPool.Put(buf)
 	return
 }
+
+// BufReaderAt wraps a byte array with the ReaderAt interface
+type BufReaderAt struct {
```

Comment:
`bytes.Reader` already providers a `ReaderAt` interface, and `Size` on a byte slice, so this seems unnecessary.

Reply:
Dropped this patch now.

Reply:
Also, I 'downgraded' the parameters to the encryption function to not take a `content.ReaderAt` but a `io.ReaderAt` instead. 

In a tree that has all the image encryption code I tried to then convert the `io.ReaderAt` to a plain `io.Reader` but that doesn't fit with the function for reading (layer blobs) from the `content.Store` since this one returns a `ReaderAt` interface via the `content.Provider` interface. I suppose we can leave it like this unless we somehow wrap that `content.ReaderAt` with some sort of helper that makes it looks like `io.Reader` interface.